File: tests/co-simulations/static/0/python/tb.py
import sys
import traceback

import cosim
import vfbdb


WRITE_FIFO_PATH = sys.argv[1]
READ_FIFO_PATH = sys.argv[2]

CLK_PERIOD = 10

iface = cosim.Iface(WRITE_FIFO_PATH, READ_FIFO_PATH)

try:
    Main = vfbdb.Main(iface)

    id = Main.ID.read()
    assert id == Main.ID.value, f"Read wrong ID {id}, expecting {Main.ID.value}"
    print(f"ID: {id}\n")

    ts = Main.TIMESTAMP.read()
    assert ts == Main.TIMESTAMP.value, f"Read wrong TIMESTAMP {ts}, expecting {Main.TIMESTAMP.value}"
    print(f"Timestamp: {ts}\n")

    iface.end(0)

except Exception as E:
    iface.end(1)
    print(traceback.format_exc())

File: tests/co-simulations/static/0/c-sync/tb.c
#include <assert.h>
#include <stdio.h>

#include "cosim_iface.h"

#include "vfbdb.h"
#include "Main.h"
#define VFBDB_IFACE &iface


int main(int argc, char *argv[]) {
	assert(argc == 3);

	vfbdb_iface_t iface = cosim_iface_iface();

	cosim_iface_init(argv[1], argv[2], NULL);

	uint32_t id;
	vfbdb_read(Main_ID, &id);
	if (id != vfbdb_Main_ID) {
		fprintf(stderr, "read wrong ID %x, expecting %x\n", id, vfbdb_Main_ID);
		cosim_iface_end(1);
	}

	uint32_t timestamp;
	vfbdb_read(Main_TIMESTAMP, &timestamp);
	if (timestamp != vfbdb_Main_TIMESTAMP) {
		fprintf(stderr, "read wrong TIMESTAMP %x, expecting %x\n", id, vfbdb_Main_TIMESTAMP);
		cosim_iface_end(1);
	}

	cosim_iface_end(0);

	return 0;
}
